.github/workflows/ci.yml

```diff
@@ -0,0 +1,192 @@
+  # Cancel intermediate builds: always.
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+
+jobs:
+  linting:
+    name: Linting checks with pre-commit
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ucl/composite-actions/pre-commit@v1
+
+  build:
+    name: ${{ matrix.build_type }} build on ${{ matrix.os }} with build testing ${{ matrix.build_testing }}
+    needs: linting
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        build_type: [Debug]
+        # Only Debug builds (so we have helpful debug output in our logs). Can
+        # also test 'Release'.
+        build_testing: [ON, OFF]
+        exclude:
+          # Currently our tests don't compile in Windows so skip this.
+          - os: windows-latest
+            build_testing: ON
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up MATLAB
+        uses: matlab-actions/setup-matlab@v1.2.3
+
+      # -------------------------------------------------------------------------------
+      # Ubuntu
+      - name: Install dependencies for Ubuntu
+        if: ${{ contains(matrix.os, 'ubuntu') }}
+        run: |
+          sudo apt-get update
+          sudo apt-get install libfftw3-dev libgomp1 python3
+
+      # -------------------------------------------------------------------------------
+      # Windows
+      - name: Install miniconda for Windows
+        if: ${{ contains(matrix.os, 'windows') }}
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          auto-update-conda: true
+          python-version: 3.9
+
+      - name: Install dependencies for Windows
+        if: ${{ contains(matrix.os, 'windows') }}
+        shell: pwsh
+        run: |
+          conda install fftw --yes
+          echo "FFTWDIR=C:\Miniconda\envs\test\Library" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf8 -Append
+
+      # -------------------------------------------------------------------------------
+      # MacOS
+      - name: Install dependencies for MacOS
+        if: ${{ contains(matrix.os, 'macos') }}
+        run: brew install fftw
+
+      - name: Fix omp headers not linked on MacOS
+        if: ${{ contains(matrix.os, 'macos') }}
+        run: |
+          brew unlink libomp
+          brew link --force libomp
+          export DOMP_ROOT_MAC=-DOMP_ROOT=$(brew --prefix libomp)
+          echo "DOMP_ROOT_MAC=${DOMP_ROOT_MAC}" >> $GITHUB_ENV
+
+      # -------------------------------------------------------------------------------
+      - name: Create build environment
+        run: |
+          cmake -E make_directory ${{ runner.workspace }}/build
+
+      - name: Compile with Werror on Ubuntu and MacOS (not Windows)
+        if: ${{ contains(matrix.os, 'macos') }} || ${{ contains(matrix.os, 'ubuntu')}}
+        run: echo "CXXFLAGS=-Werror" >> $GITHUB_ENV
```

Comment:
Does this, plus the existing linting, now resolve #41 too?

Reply:
Almost, yeah. We should fix whatever's the problem with Windows first though.
I have a Windows machine and VM so I can debug if you like (unless you have a partition?).